src/TaskLineRenderer.ts

```diff
@@ -269,49 +300,19 @@ function getTaskComponentClass(component: TaskLayoutComponent, task: Task) {
 function getComponentDataAttribute(component: TaskLayoutComponent, task: Task) {
     const dataAttribute: AttributesDictionary = {};
 
-    // If a TaskLayoutComponent needs a data attribute in the task's <span>, add the data attribute name
-    // to this dictionary: key is the component, value is the data attribute name.
-    // Otherwise, just leave an empty string ('') as the value.
-    // Also add the new component to the switch-case below in this function. This is where
-    // the data attribute value shall be calculated and set in the returned dictionary.
-    const DataAttributeNames: { [c in TaskLayoutComponent]: string } = {
-        createdDate: 'taskCreated',
-        dueDate: 'taskDue',
-        startDate: 'taskStart',
-        scheduledDate: 'taskScheduled',
-        doneDate: 'taskDone',
-        priority: 'taskPriority',
-        description: '',
-        recurrenceRule: '',
-        blockLink: '',
-    };
-
-    switch (component) {
-        case 'description':
-        case 'recurrenceRule':
-        case 'blockLink':
-            break;
-        case 'priority': {
-            const attributeName = DataAttributeNames[component];
-            dataAttribute[attributeName] = PriorityTools.priorityNameUsingNormal(task.priority).toLocaleLowerCase();
-            break;
-        }
-        case 'createdDate':
-        case 'dueDate':
-        case 'startDate':
-        case 'scheduledDate':
-        case 'doneDate': {
-            const date = task[component];
-            if (date) {
-                const attributeValue = dateToAttribute(date);
-                if (attributeValue) {
-                    const attributeName = DataAttributeNames[component];
-                    dataAttribute[attributeName] = attributeValue;
-                }
-            }
-            break;
+    // If a TaskLayoutComponent needs a data attribute in the task's <span>, get the data attribute name (key) &
+    // data attribute value (value). Otherwise, just leave an empty string ('') as the value.
+    // The value is calculated based on FieldLayoutDetail.attributeValueCalculator
+    const fieldLayoutDetail = FieldLayouts[component];
+    if (fieldLayoutDetail) {
```

Comment:
Oooh - I wonder if this line 307 is like the fall-through case in the previous `switch` statements...

Under what circumstances would this be falsey?

If it is because somebody forgot to add a new value to `FieldLayouts`, then is this a potential pitfall in maintenance?

Same applies to line 290.

Reply:
For now `component: TaskLayoutComponent` and 

```ts
export type TaskLayoutComponent =
    | 'description'
    | 'priority'
    | 'recurrenceRule'
    | 'createdDate'
    | 'startDate'
    | 'scheduledDate'
    | 'dueDate'
    | 'doneDate'
    | 'blockLink';
```

so nothing actually is preventing adding a `abrakadabra` to `TaskLayoutComponent` and then calling `Task[abrakadabra]`. So for now I think it is better to keep it this way just as a fail safe.

I think later we could make `export type TaskLayoutComponent = keyof Task` and then this shall be removed.

Reply:
I tried adding a new value to `TaskLayoutComponent` for a field that was already in Task - and did indeed get one error - that the value was missing from `FieldLayouts`.

Thank you.

Resolving.